pow/src/lib.rs

```diff
@@ -41,14 +41,23 @@ pub mod app {
 }
 
 /// Checks whether the given hash is above difficulty.
-fn is_valid_hash(hash: &H256, difficulty: Difficulty) -> bool {
+pub fn is_valid_hash(hash: &H256, difficulty: Difficulty) -> bool {
 	let num_hash = U256::from(&hash[..]);
 	let (_, overflowed) = num_hash.overflowing_mul(difficulty);
 
 	!overflowed
 }
 
-fn key_hash<B, C>(
+pub fn hash_diff(hash: &H256) -> U256 {
```

Comment:
What is hash_diff exactly and is it used anywhere?

Reply:
It is not used internally, I included it in hopes to make `kulupu_pow` more useful as a library outside of the node implementation.

It calculates the hash difficulty and is AFAIK correct.

Reply:
I don't think there's a concept of "difficulty of a hash", at least not right now in the code. We only need the functionality of checking whether a hash is valid given a difficulty `is_valid_hash`.

I hope you can explain your usecase more on why you need `hash_diff`. Otherwise I'd recommend not adding it, as it introduces unnecessary maintenance cost (we always need to keep `is_valid_hash` and `hash_diff` consistent but one of the function is not used).

Reply:
It's used in a mining environment. I might just remove it though. I agree with all of your points.